inc/class-sitemaps.php

```diff
@@ -1350,22 +1350,36 @@ public function user_sitemap_remove_excluded_authors( $users ) {
 			foreach ( $users as $user_key => $user ) {
 				$exclude_user = false;
 
-				$is_exclude_on = get_the_author_meta( 'wpseo_excludeauthorsitemap', $user->ID );
-				if ( $is_exclude_on === 'on' ) {
-					$exclude_user = true;
+				/**
+				 * Cheapest condition first; we have all information already.
+				 */
+				if ( ! $exclude_user ) {
+					$user_role    = $user->roles[0];
+					$target_key   = "user_role-{$user_role}-not_in_sitemap";
+					$exclude_user = isset( $options[ $target_key ] ) && true === $options[ $target_key ];
+					unset( $user_role, $target_key );
 				}
-				elseif ( $options['disable_author_noposts'] === true ) {
+
+				/**
+				 * If the author has been excluded by preference on profile.
+				 */
+				if ( ! $exclude_user ) {
```

Comment:
Because it is a completely different condition which should only be performed when the user is not excluded after the first condition.
It removes the need to do a database query in the `get_the_author_meta` if the users is already excluded generally by role.